### PR TITLE
Typescript error busting - contributionsLandingInit.ts

### PR DESCRIPTION
**Pull request by TomPretty**
## What are you doing in this PR?

Fix the errors in `contributionsLandingInit.ts`.

### CODE REVIEW

File: support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts [hunk 1]
```diff
@@ -49,6 +50,8 @@ import {
 } from './contributionsLandingActions';
 import type { State } from './contributionsLandingReducer';
 
+type Action = ContribAction | CommonAction;
+
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts:54]: Is there a better way of doing this? I don't think I've actually had to fix a file before that dispatched actions for reducers defined in multiple files. Here I've had to combine the common & contributionsLanding reducer actions into a single type.

File: support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts [hunk 2]
```diff
@@ -261,8 +266,8 @@ function selectInitialContributionTypeAndPaymentMethod(
 	return contributionType;
 }
 
-const init = (store: Store<State, Action, (...args: any[]) => any>) => {
-	const { dispatch } = store;
+const init = (store: Store<State, Action>): void => {
+	const dispatch = store.dispatch as ThunkDispatch<State, void, Action>;
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts:270]: I'm not sure if there's a better way of doing this. The `store.dispatch` is of type `Dispatch<Action>`, but we want it to be a `ThunkDispatch`. I couldn't easily find a way of getting that into the `Store<State, Action>` type signature

File: support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts (hunk 1 above)
Comment (i-hardy) [reply support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts:54]: Type issues around `Dispatch` and `Action` are one of the things Redux Toolkit will help us with- it essentially enables you to [infer types for these things from your base store](https://redux-toolkit.js.org/usage/usage-with-typescript). I agree that this is pretty clunky, but I think it's clear and functional enough that it's probably not worth spending more time on if we're going to be adopting Toolkit in the near future.

**Review by i-hardy — APPROVED** [review]
👍

File: support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts (hunk 1 above)
Comment (TomPretty) [reply support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts:54]: Oooo just read through the bit about getting the types for `Dispatch` and `Action` and that looks great! Very excited that we're working on adopting this

**Pull request merged**